<!-- 
Example launch file: launches the scan matcher with pre-recorded data
-->

<launch>

#### Playback ROSBAG  ##############################


<!--
<arg name="map_path"      value="$(find map_server)/maps/carters_penstock.pcd" />
<arg name="bagfile_path"  value="$(env ROS_DATASET_PATH)/13Jan_khex_statefarm/1.bag" />
<arg name="bagfile_path"  value="$(env ROS_DATASET_PATH)/13Jul2015TowneBase/towne06.bag" />
<arg name="map_path"      value="$(find map_server)/maps/towne_basement.pcd" />
<arg name="map_path"      value="$(find map_server)/maps/glen_penstock_inclined.pcd" />
<arg name="bagfile_path"  value="$(env ROS_DATASET_PATH)/glen_canyon_22July2015_after_lunch/06.bag" />
-->

# 3 needs work
# 16.bag is the inclination
# Problem in glen_penstock_incline/01.bag occurs at [89-91] sec.
# *** Show after lunch 04.bag [start 70] and 06.bag

<arg name="bagfile_path"  value="$(env ROS_DATASET_PATH)/glen_canyon/monocular/07.bag" />
<arg name="map_path"      value="$(find map_server)/maps/glen_penstock_horizontal.pcd" />

<!-- carters_gazebo/1.bag from 122 for 3 secs. works well -->

<node	pkg ="rosbag"
		name  ="rosbag"
		type  ="play"
    output="screen"
    args  ="--loop --rate=0.55 $(arg bagfile_path) --start=10 --duration=9999 --delay=5 ">
    <remap from="/control_odom" to="/dummy1" />
    <remap from="/tunnel_localizer/odom_out" to="dummy2" />
    <!--
        <remap from="/raw_imu" to="/khex/imu" />
    <remap from="/raw_imu" to="/khex/imu" />
    <remap from="/cam0/image/compressed" to="/cam2/image" />
    <remap from="/cam1/image/compressed" to="/cam0/image" />
    -->
</node>

#### start the Tunnel Localizer ##############################

<node	pkg    = "tunnel_localizer" 
		name   = "tunnel_localizer" 
		type   = "tunnel_localizer" 
		output = "screen">

		<remap from="~imu"			    to="/khex_com/imu"/>
		<remap from="~scan/top"		  to="/scan/top"/>
		<remap from="~scan/bottom"	to="/scan/bottom"/>
		<remap from="~odom_in"			to="/control_odom"/>
		<remap from="~cam/top"		  to="/cam1/image"/>
		<remap from="~cam/bottom"	  to="/cam2/image"/>
		<remap from="~cam/right"	  to="/cam0/image"/>
		<remap from="~cam/left"		  to="/cam3/image"/>

		<param name="debug_mode"	value="FALSE"/>
		<param name="refresh_rate"	value="40.0"/>	
		<param name="map_path"		value="$(arg map_path)"/>
		<param name="top_lidar_calib_file"		  value="$(find calib_data)/lidar/ins_khex_top_lidar.yaml"/>
		<param name="bottom_lidar_calib_file"	  value="$(find calib_data)/lidar/ins_khex_bottom_lidar.yaml"/>
		<param name="top_cam_calib_file"        value="$(find calib_data)/camera/ins_khex_top_cam.yaml"/>
		<param name="right_cam_calib_file"		  value="$(find calib_data)/camera/ins_khex_right_cam.yaml"/>
		<param name="bottom_cam_calib_file"		  value="$(find calib_data)/camera/ins_khex_bottom_cam.yaml"/>
		<param name="left_cam_calib_file"		    value="$(find calib_data)/camera/ins_khex_left_cam.yaml"/>
</node> 

#### start UKF Lite ##############################

<node pkg ="quadrotor_ukf_lite" 
	  name="quadrotor_ukf" 
	  type="quadrotor_ukf" 
	  output="screen">
      <rosparam file="$(find quadrotor_ukf_lite)/config/params.yaml"/>
      <remap from="~imu" to="/khex_com/imu"/> 
      <remap from="~odom_slam" to="/tunnel_localizer/odom_out"/>
</node> 

#### start RVIZ Visualization ##############################

<node	pkg ="odom_visualization"
		name="meas_odom_visualization"
		type="odom_visualization_node"
		output="screen">
		<remap from="~odom" to="/tunnel_localizer/odom_out"/>
	    <rosparam file="$(find odom_visualization)/config/config01.yaml" />
</node>

<node	pkg ="odom_visualization"
		name="ukf_odom_visualization"
		type="odom_visualization_node"
		output="screen">
		<remap from="~odom" to="/control_odom"/>
	    <rosparam file="$(find odom_visualization)/config/config02.yaml" />

</node>

<node	pkg ="odom_visualization"
		name="goal_odom_visualization"
		type="odom_visualization_node"
		output="screen">
		<remap from="~odom" to="/rc_nav/goal"/>
	    <rosparam file="$(find odom_visualization)/config/config03.yaml" />

</node>

<node pkg    = "status_visualization"
        name   = "status_visualization"
        type   = "status_visualization_node"
        output = "screen" >

        <remap from="~rc"     to="/khex_com/rc" />
        <remap from="~status" to="/khex_com/status" />
        <rosparam file="$(find status_visualization)/config/ins_khex_status_config.tunnel_localizer.yaml" />
        <param name="refresh_rate"   value="25.0" />
        <param name="debug_mode"     value="FALSE" />
        <param name="rc_config_path" value="$(find rc_proc)/config/ins_khex_rc_config.yaml" />
        <param name="panel_path"     value= "$(find status_visualization)/config/panel.png" />
</node>

<node	pkg="rviz"
		name="rviz"
		type="rviz"
		output="screen"
		args  ="-d $(find tunnel_localizer)/rviz/tunnel_localizer01.rviz">
		<param name="fixed-frame"	 value="world"/>
</node>

<node pkg="tf" type="static_transform_publisher" name="world_to_map" args="0 0 0 0 0 0 /map /world 100"/>

<!--
<node pkg  = "image_transport"
      name = "republish0"
      type = "republish"
      output = "screen"
      args = " compressed in:=/cam0/image raw out:=/cam2/image" >
</node>

<node pkg  = "image_transport"
      name = "republish1"
      type = "republish"
      output = "screen"
      args = " compressed in:=/cam1/image raw out:=/cam0/image" >
</node>
-->



</launch>















